clab/general.go

```diff
@@ -0,0 +1,127 @@
+package clab
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/srl-labs/containerlab/types"
+)
+
+var (
+	CLAB_HOSTENTRY_PREFIX  = "###### CLAB-%s-START ######"
+	CLAB_HOSTENTRY_POSTFIX = "###### CLAB-%s-END ######"
+)
+
+func AppendHostsFileEntries(containers []types.GenericContainer, labname string) error {
+	if labname == "" {
+		return fmt.Errorf("missing lab name")
+	}
+	// lets make sure we do not have remaining of a non destroyed run in the hosts file
+	err := DeleteEntriesFromHostsFile(labname)
+	if err != nil {
+		return err
+	}
+	data := GenerateHostsEntries(containers, labname)
+	if len(data) == 0 {
+		return nil
+	}
+	f, err := os.OpenFile("/etc/hosts", os.O_APPEND|os.O_WRONLY, os.ModeAppend)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+	_, err = f.WriteString("\n")
+	if err != nil {
+		return err
+	}
+	_, err = f.Write(data)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// hostEntries builds an /etc/hosts compliant text blob (as []byte]) for containers ipv4/6 address<->name pairs
+func GenerateHostsEntries(containers []types.GenericContainer, labname string) []byte {
+	buff := bytes.Buffer{}
+	v4buff := bytes.Buffer{}
```

Comment:
Do you need the `v4buff` if you write the host entry prefix and IPv4 to `buff` and only have one other intermediary `v6buff` to write after the loop?

Reply:
thought it might be nice to collect everything up front and finally compose it, just for readability.
However compressed it now.